worlds/kh1/Rules.py

```diff
@@ -53,487 +81,570 @@ def has_all_magic_lvx(state: CollectionState, player: int, level) -> bool:
+        or (logic_difficulty > 10 and state.has("Progressive Stop", player))
+    )
 
-def has_reports(state: CollectionState, player: int, eotw_required_reports: int) -> bool:
-    return state.has_group_unique("Reports", player, eotw_required_reports)
+def has_lucky_emblems(state: CollectionState, player: int, required_amt: int) -> bool:
+    return state.has("Lucky Emblem", player, required_amt)
 
-def has_final_rest_door(state: CollectionState, player: int, final_rest_door_requirement: str, final_rest_door_required_reports: int, keyblades_unlock_chests: bool, puppies_choice: str):
-    if final_rest_door_requirement == "reports":
-        return state.has_group_unique("Reports", player, final_rest_door_required_reports)
-    if final_rest_door_requirement == "puppies":
-        if puppies_choice == "individual":
-            return has_puppies_individual(state, player, 99)
-        if puppies_choice == "triplets":
-            return has_puppies_triplets(state, player, 99)
-        return has_puppies_all(state, player, 99)
-    if final_rest_door_requirement == "postcards":
-        return state.has("Postcard", player, 10)
-    if final_rest_door_requirement == "superbosses":
-        return (
-                state.has_all({"Olympus Coliseum", "Neverland", "Agrabah", "Hollow Bastion", "Green Trinity", "Phil Cup", "Pegasus Cup", "Hercules Cup", "Entry Pass"}, player)
-                and has_emblems(state, player, keyblades_unlock_chests)
-                and has_all_magic_lvx(state, player, 2)
-                and has_defensive_tools(state, player)
-                and has_x_worlds(state, player, 7, keyblades_unlock_chests)
-            )
+def has_final_rest_door(state: CollectionState, player: int, final_rest_door_requirement: str, final_rest_door_required_lucky_emblems: int):
+    if final_rest_door_requirement == "lucky_emblems":
+        return state.has("Lucky Emblem", player, final_rest_door_required_lucky_emblems)
+    else:
+        return state.has("Final Door Key", player)
 
-def has_defensive_tools(state: CollectionState, player: int) -> bool:
-    return (
+def has_defensive_tools(state: CollectionState, player: int, logic_difficulty: int) -> bool:
+    if logic_difficulty > 14:
+        return True
+    else:
+        return (
             state.has_all_counts({"Progressive Cure": 2, "Leaf Bracer": 1, "Dodge Roll": 1}, player)
             and state.has_any_count({"Second Chance": 1, "MP Rage": 1, "Progressive Aero": 2}, player)
         )
 
+def has_basic_tools(state: CollectionState, player: int) -> bool:
+    return (
+            state.has_all_counts({"Dodge Roll": 1, "Progressive Cure": 1}, player)
+            and state.has_any_count({"Combo Master": 1, "Strike Raid": 1, "Sonic Blade": 1, "Counterattack": 1}, player)
+            and state.has_any_count({"Leaf Bracer": 1, "Second Chance": 1, "Guard": 1}, player)
+            and has_offensive_magic(state, player, 6)
+        )
```

Comment:
```suggestion
def has_basic_tools(state: CollectionState, player: int) -> bool:
    return (
            state.has_any({"Dodge Roll", "Progressive Cure"}, player)
            and state.has_any({"Combo Master", "Strike Raid", "Sonic Blade", "Counterattack"}, player)
            and state.has_any({"Leaf Bracer", "Second Chance", "Guard"}, player)
            and has_offensive_magic(state, player, 6)
        )
```
No real need to use `has_any_count` if the count for all of them is 1